### ISSUE TITLE
Calibra-Web fails to “Read in Browser” some ePub on iOS

### ISSUE THREAD

**Issue by Smithoo4** [janeczku/calibre-web#1284]
Calibra-Web fails to “Read in Browser” some ePub on iOS

On some ePub when I try to “Read in Browser” on my iPhone or iPad Calibra-Web will open a new tab/page as expected but the content will not load. The loading circle will continually move but nothing happens, or it will take a long time (2 min) and I have to make sure the device does no go to sleep. I don’t have this issue when I try to “Read in Browser” on a desktop. I only seem to have this issue with larger ePub (~240 MB) with lots of pictures, mostly cookbooks.

Below is the Calibre-Web Log with debug on well I try to open one of these ePub on my iPhone.

```
[2020-03-31 15:44:32,524] DEBUG {cps.web:1395} Start epub reader for 582
[2020-03-31 15:44:32,617]  INFO {cps.web:1033} Serving book: Japanese Home Cooking - Sonoko 
Sakai 

```
And here is my system info:

```
Program Library     Installed Version
Platform            Linux 5.4.28-1-MANJARO #1 SMP PREEMPT Wed Mar 25 12:45:29 UTC 2020 x86_64
Python              3.8.2 (default, Feb 26 2020, 22:21:03) [GCC 9.2.1 20200130]
Calibre_Web         0.6.7 Beta - $Format:%H$ - $Format:%cI$
WebServer           Tornado 6.0.4
Flask               1.1.1
Flask_Login         0.5.0
Flask_Principal     0.4.0
Werkzeug            1.0.0
Babel               2.8.0
Jinja2              2.11.1
Requests            2.23.0
SqlAlchemy          1.3.15
pySqlite            2.6.0
SQLite              3.31.1
iso639              0.4.5
pytz                2019.3
Unidecode           installed
Flask_SimpleLDAP    not installed
Goodreads           not installed
jsonschema          not installed
Image Magick        ImageMagick 7.0.10-2 Q16 x86_64 2020-03-22 https://imagemagick.org
PyPdf               v1.26.0
lxml                not installed
Wand                0.5.9
Pillow              v6.2.1
Comic_API           not installed
ebook converter     not configured
```
Is there anything I can do to get these ePub to load on iOS?

Thank you

**Comment by OzzieIsaacs** [comment]
Thanks for the detailed feedback. I'll try to reproduce the feature. I think that you are on a good track with the book size. 
The epub reader is an external library with a lot of issues on the github page and the issues are mainly (not like here where many issues are feature requests) problems/bugs. We already opened one  or two issues, but never got feedback. 
Making long story short: I don't think I can fix it

Do the books open on a normal desktop/notebook computer?
Could you please send me one of the non working books (via private email from my profile page) , so I can check if I can reproduce it?

**Comment by Smithoo4** [comment]
Thank you for looking into this, I fully understand if you are not able to fix this.

This issue only seem to be on iPhones and iPads (Chrome/safari). The book will open correctly on desktop/notebook computer (Chrome/Firefox) but do take their time.

I will send you one of the books.

Thank you

**Comment by hallamjeff** [comment]
I don't think it is isolated to iPhones and iPads. I am experiencing the same issue trying to read a CBZ file in the browser on an Android tablet. The file is 198MB. It loads ok on a desktop.
